rust/lance/src/dataset/take.rs

```diff
@@ -131,6 +129,36 @@ async fn do_take_rows(
 ) -> Result<RecordBatch> {
     let with_row_id_in_projection = projection.physical_projection.with_row_id;
     let with_row_addr_in_projection = projection.physical_projection.with_row_addr;
+    let physical_schema = Arc::new(projection.physical_projection.to_bare_schema());
+
+    // short cut: if projection do not contain any physical column, directly returns id & addr columns
+    if physical_schema.fields.is_empty()
+        && builder.row_ids.is_some()
+        && (with_row_id_in_projection || with_row_addr_in_projection)
```

Comment:
Can we test what happens in these cases?

```
        let projection = ProjectionRequest::from_columns(["_rowid"], dataset.schema());
        let projection = ProjectionRequest::from_columns(["_rowaddr"], dataset.schema());
```

The first one is admittedly, a little silly, the user is asking for _rowid and passing in _rowid but users do silly things sometimes :shrug:.  I am worried we might be returning `_rowaddr` even though the user didn't ask for it.

In the second one maybe the user is just trying to convert one set of `_rowid` values into a set of `_rowaddr` values and doesn't care about the ordering / mapping.  Again, I am worried we might be returning `_rowid` even though the user didn't ask for it.